crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2270,98 +2291,132 @@ impl AuthorityPerEpochStore {
             .db_batch()
             .expect("Consensus should not be processed past end of epoch");
 
-        // Pre-process transactions to find the most recent randomness round included in the commit.
-        let mut last_randomness_round_written = self.last_randomness_round_written()?;
-        // There must be at most one RandomnessStateUpdate per commit.
-        let mut randomness_state_update_found = None;
-        for tx in system_transactions.iter() {
-            let SequencedConsensusTransactionKind::System(tx) = &tx.0.transaction else {
-                unreachable!("system_transactions vector should only contain system transactions")
-            };
-            if let TransactionKind::RandomnessStateUpdate(rsu) =
-                tx.data().intent_message().value.kind()
-            {
-                assert!(
-                    randomness_state_update_found.is_none(),
-                    "found multiple RandomnessStateUpdates in one commit: {:?}, {rsu:?}",
-                    randomness_state_update_found.unwrap(),
-                );
-                randomness_state_update_found = Some(rsu);
-                last_randomness_round_written = std::cmp::max(
-                    last_randomness_round_written,
-                    RandomnessRound(rsu.randomness_round),
-                );
-            }
-        }
+        // Save roots for checkpoint generation. One set for most tx, one for randomness tx.
+        let roots: BTreeSet<_> = system_transactions
+            .iter()
+            .chain(sequenced_transactions.iter())
+            // no need to include end_of_publish_transactions here
```

Comment:
Could you comment on why we don't need to include end_of_publish_transactions?

Reply:
done